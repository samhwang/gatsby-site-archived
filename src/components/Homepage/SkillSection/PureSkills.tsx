import Skill from './Skill';
import LangIcon from '../../Icons/LangIcon';
import type { SkillsMetadata, TechStackMetadata } from '../../../globals';

interface PureSkillsProps {
  skills: SkillsMetadata;
  techStack: TechStackMetadata;
}

function PureSkills({ skills, techStack }: PureSkillsProps) {
  return (
    <section
      className="resume-section p-3 p-lg-5 d-flex align-items-center"
      id="skills"
    >
      <div className="w-100">
        <h2 className="mb-5">Skills and Proficiency</h2>

        {skills.map(({ title, description }) => (
          <Skill title={title} description={description} key={title} />
        ))}
        <ul className="list-inline dev-icons">
          {techStack.map((language) => (
            <LangIcon name={language} key={language} />
          ))}
        </ul>
      </div>
    </section>
  );
}

export default PureSkills;
