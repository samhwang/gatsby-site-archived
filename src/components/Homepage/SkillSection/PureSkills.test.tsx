/* eslint-disable import/no-extraneous-dependencies */
import renderer from 'react-test-renderer';
import PureSkills from './PureSkills';

describe('Skill Section Component', () => {
  it('Should render without crashing and match snapshot', () => {
    const skills = [
      {
        title: 'asdfasdf',
        description: ['asdfasdf1234'],
      },
    ];
    const techStack = ['js'];

    const tree = renderer
      .create(<PureSkills skills={skills} techStack={techStack} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
