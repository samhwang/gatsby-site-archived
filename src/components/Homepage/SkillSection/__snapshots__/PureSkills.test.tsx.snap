// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skill Section Component Should render without crashing and match snapshot 1`] = `
<section
  className="resume-section p-3 p-lg-5 d-flex align-items-center"
  id="skills"
>
  <div
    className="w-100"
  >
    <h2
      className="mb-5"
    >
      Skills and Proficiency
    </h2>
    <div
      className="subheading mb-3"
    >
      asdfasdf
    </div>
    <ul>
      <li>
        asdfasdf1234
      </li>
    </ul>
    <ul
      className="list-inline dev-icons"
    >
      <li
        className="list-inline-item"
      />
    </ul>
  </div>
</section>
`;
